specs/pluggable_alert_delivery.adoc

```diff
@@ -0,0 +1,371 @@
+  'node-id': <id of node on which alert was detected>,
+  'time-stamp': <time stamp of alert>,
+  'resource': <the name of resource for which alert has been raised>,
+  'current-value': <the current observed value status/utilization as applies>
+  'tags': <custom alert specific info>,
+  'type': <the type of alert percent-used/status of resource>,
+  'severity': <severity of alert>,
+  'significance': <the severity of importance of notifying the alert>,
+  'ackedby': <indicate who acked the alert>,
+  'acked': <boolean to indicate if the alert is acked>,
+}
+----
+
+ex:
+For a performance monitoring related alert:
+
+----
+{
+  alert_id : '6405962e-bc46-11e6-a4a6-cec0c932ce01',
+  node_id: '5205962e-bc46-11e6-a4a6-cec0c932cz01',
+  time_stamp: '1481046935.536',
+  'resource': 'memory',
+  'CurrentValue': '7.176942e+00',
+  tags: {
+    'WarningMax': '1.000000e+00',
+    'FailureMax': '2.000000e+00',
+  },
+  'Type': 'percent-used',
+  'severity': 'Critical',
+  'ackedby': '',
+  'acked': False,
+  'significance': 'HIGH',
+}
+----
+
+and for a status based alert:
+
+----
+{
+  alert_id : '6405962e-bc46-11e6-a4a6-cec0c932ce01',
+  node_id: '5205962e-bc46-11e6-a4a6-cec0c932cz01',
+  time_stamp: '1481046935.536',
+  'resource': 'cluster',
+  'CurrentValue': 'Down',
+  tags: {
+    'cluster_id' : '6406062e-be46-11e6-a4a6-cec0c932ce01',
+  },
+  'Type': 'status',
+  'severity': 'Critical',
+  'ackedby': '',
+  'acked': False,
+  'significance': 'HIGH',
+}
+----
+
+The severity levels can be 'Critical', 'Info' or 'Warning'.
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+The tendrl api needs to proxy to apis exposed by Tendrl/alerting as mentioned
+in section below(Notifications/Monitoring impact)
+
+==== Notifications/Monitoring impact:
+
+Tendrl/alerting needs to implement flows for apis as described in section
+"Tendrl API impact".
+
+The flow definition for the above will look like:
+
+----
+# flake8: noqa
+data = """---
+namespace.tendrl.alerting:
+  objects:
+    Alert:
+      attrs:
+        alert-id:
+          type: String
+        node-id:
+          type: String
+        time-stamp:
+          type: String
+        resource:
+          type: String
+        current-value:
+          type: String
+        tags:
+          type: String
+        type:
+          type: String
+        severity:
+          type: String
+        significance:
+          type: String
+        ackedby:
+          type: String
+        acked:
+          type: Boolean
+      enabled: true
+      value: alerts/$Alert.alert_id
+      list: alerts/
+        filter_criteria:
+          type: json
+    NotificationMedia:
+      attrs:
+        name:
+          type: String
+        list: alerts/notification_medium/
+tendrl_schema_version: 0.3
+"""
+----
+
+* This adds the following apis:
+  ** Api to get list of currently supported means of notification.
+
+----
+GET /alerting
+
+Sample Response:
+
+Status: 200 OK
+{
+  email,
+  snmp
```

Comment:
Ok updating it to :

GET /alerting/supported_notification_medium

Sample Response:

Status: 200 OK
{
  notif_medium: [email, snmp]
}

Is the above ok?